SoundSwitch/Framework/NotificationManager/MMNotificationClient.cs

```diff
@@ -51,14 +51,14 @@ public void OnDeviceRemoved(string deviceId)
 
         public void OnDefaultDeviceChanged(DataFlow flow, Role role, string defaultDeviceId)
         {
-            Task.Factory.StartNew(() =>
+            if (defaultDeviceId != null)
```

Comment:
Can you invert the condition and do return ?
That what I meant with early exit.

```csharp
if (defaultDeviceId == null)
    return;
```

I find it easier to read code when you know the case that are exiting directly a method.

Reply:
Ah sry I misunderstood. Done now. 😄

(I squashed the commits to make it look cleaner.)